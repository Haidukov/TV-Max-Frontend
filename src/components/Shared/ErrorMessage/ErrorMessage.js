import React from 'react';
import PropTypes from 'prop-types';

import s from './ErrorMessage.scss';

const ErrorMessage = ({ children }) => (
  <div className={s.error} role='alert'>{children}</div>
);

ErrorMessage.propTypes = {
  children: PropTypes.node.isRequired
};

export default ErrorMessage;
