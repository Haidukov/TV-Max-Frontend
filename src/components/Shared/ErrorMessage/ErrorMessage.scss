@import '../../../assets/styles/variables.scss';

.error {
    display: flex;
    justify-content: center;
    margin-top: 50px;
    font-size: 20px;
    color: $red;
}