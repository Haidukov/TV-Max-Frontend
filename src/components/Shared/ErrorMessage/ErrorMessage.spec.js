import React from 'react';
import { render } from '@testing-library/react';
import ErrorMessage from './ErrorMessage';

describe('<ErrorMessage />', () => {
  it('should match snapshot', () => {
    const TestComponent = () => <div>test</div>;
    const { asFragment } = render(
      <ErrorMessage>
        <TestComponent />
      </ErrorMessage>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
