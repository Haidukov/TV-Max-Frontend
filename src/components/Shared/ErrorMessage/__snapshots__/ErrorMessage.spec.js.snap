// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="error"
    role="alert"
  >
    <div>
      test
    </div>
  </div>
</DocumentFragment>
`;
